lib/Parse/ParseDecl.cpp

```diff
@@ -1460,6 +1347,41 @@ bool Parser::parseDeclAttribute(DeclAttributes &Attributes, SourceLoc AtLoc) {
         .fixItReplace(Tok.getLoc(), "available");
   }
 
+
+  if (DK == DAK_Count && Tok.getText() == "warn_unused_result") {
```

Comment:
I think I'd rather leave the attribute kind in the list and then just not actually create one, rather than text-matching on the syntax of former attributes.